packages/x-data-grid-premium/src/tests/rowGrouping.DataGridPremium.test.tsx

```diff
@@ -86,6 +86,10 @@ describe('<DataGridPremium /> - Row grouping', () => {
   describe('Setting grouping criteria', () => {
     clock.withFakeTimers();
 
+    after(() => {
+      clock.restore();
```

Comment:
We seem to be calling it after each test already: https://github.com/mui/material-ui/blob/bddf126c978ae50dd67ffe0ff143dbb1e42eec3d/packages-internal/test-utils/src/createRenderer.tsx#L392

I'm OK with calling it in the `after` hook as well if it's less flaky with that.
But how about doing that in the `withFakeTimers` function? https://github.com/mui/material-ui/blob/bddf126c978ae50dd67ffe0ff143dbb1e42eec3d/packages-internal/test-utils/src/createRenderer.tsx#L430-L438

Reply:
I've tested that assumption in another PR and can conclude that the `after` block does not seem to get called. 🤔 
I will test it more locally with the actual package, but adding a `console.info` in the `after` block does not log anything when running a test having a describe with `withFakeTimers`.

Reply:
My assumption was not correct. The `after` hook seems to be called.
I'm currently clueless why explicitly adding the `after` hooks in the tests does seem to pass tests (ran them [3 times](https://app.circleci.com/pipelines/github/mui/mui-x?branch=pull%2F14583)) and ensuring `clock.restore` is called in the original `hook` consistently doesn't help... 🤷 
https://github.com/mui/mui-x/pull/14592

Reply:
I wasn't able to ever reproduce those flaky tests locally. 🙈 
And I also reproduced flaky tests with on this PR. 💡

Reply:
This agenda didn't produce any stable results.
I'm exploring a more long-term/stable solution. 😉